### ISSUE TITLE
[HOLD for payment 2025-01-22] [$250] System message is shown in English

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#54397]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.77-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @dukenv0307 
**Slack conversation** (hyperlinked to channel name): [expensify_bugs](https://expensify.slack.com/archives/C049HHMV9SM/p1734582221258399)

## Action Performed:
1. Change the app language to Spanish
2. Create a new workspace
3. Go to more features -> enable Company Card
4. Upgrade this workspace to the Control Plan
5. Go to workspace admin room
6. Observe system messages
## Expected Result:
System message is shown in Spanish
## Actual Result:
 System message is shown in English
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details>

https://github.com/user-attachments/assets/faa87cce-729d-44b3-8ddb-7346df90a07a

https://github.com/user-attachments/assets/7053cdf1-851f-4492-bd7d-52077e698166

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021875253474674806164</li>
        <li>Upwork Job ID: 1875253474674806164</li>
        <li>Last Price Increase: 2025-01-03</li>
<li>Automatic offers: </li>
<ul>
<li>nkdengineer | Contributor | 105614575</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @zanyrenney</details>

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

System message is shown in English

### What is the root cause of that problem?

When we upgrade the workspace to the control plan, we don't add any report action to keep track of the change

### What changes do you think we should make in order to solve the problem?
BE return action name is POLICYCHANGELOG_CORPORATE_UPGRADE
1. Create a new translation key for this action
2. Create a util function in ReportUtils like `getCorporateUpgradeMessage` and return translation
3. Use this function in some places below like we did with other actions

- Add a case for this action [here](https://github.com/Expensify/App/blob/ce282d38777e9e0a267e1b6f44bff9c4aace98ee/src/libs/OptionsListUtils.ts#L489) for the alternative text in LHN

- Add a case in `ReportActionItem` [here](https://github.com/Expensify/App/blob/f69d05b32a33d983f316d970a1d668608a42a318/src/pages/home/report/PureReportActionItem.tsx#L828) for the action in `ReportScreen`

- Add a case in [here](https://github.com/Expensify/App/blob/ce282d38777e9e0a267e1b6f44bff9c4aace98ee/src/pages/home/report/ContextMenu/ContextMenuActions.tsx#L387) for the copy message 


### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?

### What alternative solutions did you explore? (Optional)
NA

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
System message is shown in English

### What is the root cause of that problem?
We are not specially handling POLICYCHANGELOG_CORPORATE_UPGRADE type report action to provide translations so we are displaying the reportAction.message the BE returns
### What changes do you think we should make in order to solve the problem?

We should add a translation to handle translation for POLICYCHANGELOG_CORPORATE_UPGRADE action types more of similar to `CONST.REPORT.ACTIONS.TYPE.REJECTED` and create a util that returns the translation like  [here](https://github.com/FitseTLT/App/blob/582c6f14e2ae53fc4d12d355d37fa007d9796925/src/libs/ReportUtils.ts#L4866-L4867)
And then we should use this translation for:
1. For the alternative text in LHN [here](https://github.com/FitseTLT/App/blob/582c6f14e2ae53fc4d12d355d37fa007d9796925/src/libs/OptionsListUtils.ts#L598-L599) 

2. Handle the case in PureReportActionItem [here](https://github.com/FitseTLT/App/blob/582c6f14e2ae53fc4d12d355d37fa007d9796925/src/pages/home/report/PureReportActionItem.tsx#L806-L807) to display it

3. Handle the case in getReportName for chat thread header title [here](https://github.com/FitseTLT/App/blob/582c6f14e2ae53fc4d12d355d37fa007d9796925/src/libs/ReportUtils.ts#L4007-L4008) to align the thread header title with the report action content we are displaying in PureReportActionItem.
4. And finally in [here](https://github.com/Expensify/App/blob/ce282d38777e9e0a267e1b6f44bff9c4aace98ee/src/pages/home/report/ContextMenu/ContextMenuActions.tsx#L387) for copy to clipboard
### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)

**Comment by zanyrenney** [comment]
Nice find, thanks!

**Comment by zanyrenney** [comment]
thanks @dukenv0307

**Comment by FitseTLT** [comment]
Added [Notes](https://github.com/Expensify/App/issues/54397#issuecomment-2557059929) to handle a related `POLICYCHANGELOG_TEAM_DOWNGRADE` report actions on the PR phase.

**Comment by FitseTLT** [comment]
> Thanks for all your proposals.
> 
> I think we should go with @nkdengineer's [proposal](https://github.com/Expensify/App/issues/54397#issuecomment-2555953753) since it's the first and addresses more cases.
> 
> 🎀👀🎀 C+ reviewed

@dukenv0307 The selected solution misses the [getReportName](https://github.com/FitseTLT/App/blob/582c6f14e2ae53fc4d12d355d37fa007d9796925/src/libs/ReportUtils.ts#L4007-L4008) case which would create a regression because the report action would be displayed translated but the thread header title untranslated. It is an important point missed.
cc @luacmartins

**Comment by luacmartins** [comment]
@dukenv0307 thoughts on the comment above?

**Comment by FitseTLT** [comment]
> 1. It's the first proposal to point out the correct idea and places to fix. For the case [getReportName](https://github.com/FitseTLT/App/blob/582c6f14e2ae53fc4d12d355d37fa007d9796925/src/libs/ReportUtils.ts#L4007-L4008), we can address it in PR phase since it's not too hard to figure out

I don't agree @dukenv0307  Here are some issues (https://github.com/Expensify/App/issues/37292 ,https://github.com/Expensify/App/issues/34989)) where we fixed regressions caused by missing translations in copy to clipboard just similar to points missed by the first proposal so there is no guarantee that it could easily be caught on PR phase. 

> 2.It fixes 1 more place that is related to this issue.

The other point you are mentioning is in fact is not directly related to the issue in the OP and the feature of downgrading in ND is only introduced recently compared to the current issue and cannot be a main point to choose a proposal on this issue so I think that should be a point to be addressed in the PR phase.

cc @luacmartins

**Comment by luacmartins** [comment]
The proposals are almost identical, the only difference is one place that needs to be updated. Given that, I think we should just go with the 1st proposal considering this part of the contributing guidelines given the difference is pretty small:

>Note: Before submitting a proposal on an issue, be sure to read any other existing proposals. ALL NEW PROPOSALS MUST BE DIFFERENT FROM EXISTING PROPOSALS. The *difference* should be important, meaningful or considerable.

**Comment by zanyrenney** [comment]
https://github.com/Expensify/Expensify/issues/463464

**Comment by zanyrenney** [comment]
**Payment summary**

@dukenv0307 requires payment through NewDot Manual Requests - please request $250 via ND requests.
@nkdengineer requires payment [automatic offer](https://www.upwork.com/nx/wm/offer/105614575) (Contributor) - paid $250 via upwork.
